test/option-nonce.test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const jwt = require('../');
+const expect = require('chai').expect;
+const util = require('util');
+
+describe('nonce option', function () {
+  let token;
+
+  beforeEach(function () {
+    token = jwt.sign({ nonce: 'abcde' }, undefined, { algorithm: 'none' });
+  });
+  [
+    {
+      description: 'should work with string',
+      nonce: 'abcde',
+    },
+  ].forEach((testCase) => {
+    it(testCase.description, function (done) {
+      expect(jwt.verify(token, undefined, { nonce: testCase.nonce })).to.not.throw;
+      jwt.verify(token, undefined, { nonce: testCase.nonce }, (err) => {
+        expect(err).to.be.null;
+        done();
+      })
+    });
+  });
+  [
+    true,
+    'invalid',
+    [],
+    ['foo'],
+    {},
+    { foo: 'bar' },
+  ].forEach((nonce) => {
+    let tokenhoge = jwt.sign({ foo: 'bar' }, undefined, { algorithm: 'none' });
```

Comment:
You should be able to remove this line and reuse the `token` generated in the `beforeEach` above.